documents/api/viewsets.py

```diff
@@ -69,9 +70,17 @@ def get_queryset(self):
 
         return Attachment.objects.filter(**qs_filters)
 
-    @staff_required(required_permission=ServicePermissions.VIEW_ATTACHMENTS)
-    def retrieve(self, request, *args, **kwargs):
-        return super(AttachmentViewSet, self).retrieve(request, *args, **kwargs)
+    @login_required()
+    def retrieve(self, request, pk, *args, **kwargs):
+        try:
```

Comment:
Afaik, you could also replace this try-except with just `attachment = self.get_objects()`

Reply:
Ahhh nice trick! Didn't know that was possible 👍